components/omnibox/browser/suggested_sites_provider_data.cc

```diff
@@ -3,98 +3,147 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "brave/components/binance/browser/regions.h"
+#include "brave/components/ntp_widget_utils/browser/ntp_widget_utils_region.h"
 #include "brave/components/omnibox/browser/suggested_sites_provider.h"
 
 #include "base/strings/utf_string_conversions.h"
 
+namespace {
+
+const char binance_host[] = "binance.com";
+const char binance_host_full[] = "wwww.binance.com";
+const char buy_sell_path[] = "/buy-sell-crypto/";
+
+std::string GetLocalizedURL(const std::string& query_params, bool str_display) {
```

Comment:
I am having this return a string representation of the URL as we have just a display URL as a part of SuggestedSitesMatch. For that could we just do GURL.spec()? Then this function could be written cleaner building a GURL instead of a string